File: backend/src/main/java/com/runway/common/response/ApiResponse.java
package com.runway.common.response;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Getter;

@Getter
public class ApiResponse<T> {

    private final boolean success;
    private final String message;
    private final T data;

    // 에러 응답에만 포함. null이면 JSON 직렬화에서 제외
    @JsonInclude(JsonInclude.Include.NON_NULL)
    private final String errorCode;

    private ApiResponse(boolean success, String message, T data, String errorCode) {
        this.success = success;
        this.message = message;
        this.data = data;
        this.errorCode = errorCode;
    }

    public static <T> ApiResponse<T> success(String message, T data) {
        return new ApiResponse<>(true, message, data, null);
    }

    public static <T> ApiResponse<T> success(String message) {
        return new ApiResponse<>(true, message, null, null);
    }

    public static <T> ApiResponse<T> error(String message, String errorCode) {
        return new ApiResponse<>(false, message, null, errorCode);
    }
}

File: android/app/src/main/java/com/runway/android/core/model/ApiResponse.kt
package com.runway.android.core.model

/**
 * 백엔드 공통 응답 wrapper.
 *
 * 성공: { "success": true,  "message": "...", "data": {...} }
 * 실패: { "success": false, "message": "...", "errorCode": "SOME_CODE", "data": null }
 */
data class ApiResponse<T>(
    val success: Boolean,
    val message: String,
    val data: T?,
    val errorCode: String? = null,
)
